cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -608,6 +609,13 @@ private void setStreamEventDecoder(Job job) throws IOException {
     }
   }
 
+  private String getJobName(BasicMapReduceContext context) {
+    Id.Program programId = context.getProgram().getId();
+    return String.format("%s.%s.%s.%s.%s",
```

Comment:
ideally you would cover that with unit-test - so that if someone changes the format - it'll break